msteams/msteams.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+import json
+import http.client
+import urllib.parse
+import sys
+
+
+# Function to send an HTTP POST request
+def send_http_request(webhook_url, headers, payload):
+    parsed_url = urllib.parse.urlparse(webhook_url)
+    host = parsed_url.netloc
+    url_path = parsed_url.path
+
+    conn = http.client.HTTPSConnection(host)
+    conn.request("POST", url_path, body=payload, headers=headers)
+    res = conn.getresponse()
+    data = res.read()
+    conn.close()
+
+    return res.status, data.decode('utf-8')
+
+
+# Function to send message to Microsoft Teams
+def send_teams_message(webhook_url, message):
+    payload = json.dumps({
+        '@type': 'MessageCard',
+        '@context': 'http://schema.org/extensions',
+        'summary': 'RPort Notification',
+        'sections': [{
+            'activityTitle': 'RPort Notification',
+            'text': message
+        }]
+    })
+
+    headers = {'Content-Type': 'application/json'}
+    status, response_data = send_http_request(webhook_url, headers, payload)
+
+    if status == 200:
+        print(f"Successfully sent message to Teams: {message}")
+    else:
+        sys.stderr.write(f"Failed to send message to Teams. Status code: {status}, Reason: {response_data}\n")
+        exit(1)
+
+
+if __name__ == "__main__":
+    # Read JSON from stdin
+    input_json = json.load(sys.stdin)
+    message = input_json.get('data', '')
+    recipients = input_json.get('recipients', [])
+
+    if not message:
+        sys.stderr.write("No 'data' field in the input JSON or it is empty.\n")
+        exit(1)
+
+    if not recipients:
+        sys.stderr.write("No 'recipients' field in the input JSON or it is empty.\n")
+        exit(1)
+
+    for recipient in recipients:
+        # Assuming recipients are Teams webhook URLs
+        send_teams_message(recipient, message)
```

Comment:
@KonradKuznicki  We should add some minimal validation of the recipient with a regex
`^https:\/\/.*office\.com\/webhook.{0,2}\/.*\/IncomingWebhook\/`

Reply:
I think that it would be better to put in example and in docs real webhook with "x-ed" out token as an example then add validation - I think it's going to be easier to reuse - especially if user wants to cut out teams part and but something of his own